### PR TITLE
TG Port: #46425 [s] input checking

### PR DESCRIPTION
**Pull request by Toxici11i**
thanks /tg/

### CODE REVIEW

**Comment by qwertyquerty** [comment]
Zero subtlety. Also git cherry-pick exists

File: code/__DEFINES/admin.dm [hunk 1]
```diff
@@ -73,5 +73,8 @@
 #define SPAM_TRIGGER_WARNING	5	//Number of identical messages required before the spam-prevention will warn you to stfu
 #define SPAM_TRIGGER_AUTOMUTE	10	//Number of identical messages required before the spam-prevention will automute you
 
+#define MAX_KEYPRESS_COMMANDLENGTH 32 //Max length of a keypress command before it's considered to be a forged packet
+#define MAX_KEYPRESS_AUTOKICK
```
Comment (qwertyquerty) [inline code/__DEFINES/admin.dm]: Empty define?

File: code/modules/keybindings/bindings_client.dm [hunk 2]
```diff
@@ -4,33 +4,64 @@
...
+			return
+
 	keys_held[_key] = world.time
 	var/movement = SSinput.movement_keys[_key]
 	if(!(next_move_dir_sub & movement) && !keys_held["Ctrl"])
 		next_move_dir_add |= movement
 
 	// Client-level keybindings are ones anyone should be able to do at any time
 	// Things like taking screenshots, hitting tab, and adminhelps.
-	var/AltMod = keys_held["Alt"] ? "Alt-" : ""
```
Comment (qwertyquerty) [inline code/modules/keybindings/bindings_client.dm]: Dude you just completely broke our keybinds system.

Comment (Toxici11i) [reply code/modules/keybindings/bindings_client.dm]: readding

File: code/__DEFINES/admin.dm (hunk 1 above)
Comment (Toxici11i) [reply code/__DEFINES/admin.dm]: defined later I believe, or in config.

Comment (qwertyquerty) [reply code/__DEFINES/admin.dm]: No it's not defined later. Please actually check things before dismissing them

Comment (Toxici11i) [reply code/__DEFINES/admin.dm]: what do you want it to be then?

**Comment by Toxici11i** [comment]
ok its codermonky time give sec

**Comment by qwertyquerty** [comment]
Please tell me you tested this.

**Review by qwertyquerty — APPROVED** [review]
You better have tested this

**Comment by Toxici11i** [comment]
review intensively I guess

**Comment by ike709** [comment]
```
beestation.dme, line 69, column 1:
warning: duplicate #include "code/__DEFINES/metacoin.dm"
- 66:1: previously included here
```
@qwertyquerty I'm getting real sick of beecoins appearing in the travis error log.

**Comment by ike709** [comment]
Also I can assure you he hasn't thoroughly tested this.

**Review by ike709 — APPROVED** [review]

**Comment by qwertyquerty** [comment]
As soon as a PR that fixes the metacoin issue actually gets merged that problem will be no more

**Comment by qwertyquerty** [comment]
Awaiting your seal of approval @crossedfall

**Comment by Crossedfall** [comment]
> 
> 
> Awaiting your seal of approval @crossedfall

Waiting for my recent change to pass travis. This should fix the metacoin issue.

**Review by Crossedfall — APPROVED** [review]

**Comment by qwertyquerty** [comment]
What could go wrong

**Pull request merged**